tests/callbacks/Helpers.hpp

```diff
@@ -0,0 +1,145 @@
+DEFINE_EVENT_MATCHER(BeginParallelReduceEvent)
+DEFINE_EVENT_MATCHER(EndParallelReduceEvent)
+DEFINE_EVENT_MATCHER(BeginParallelScanEvent)
+DEFINE_EVENT_MATCHER(EndParallelScanEvent)
+DEFINE_EVENT_MATCHER(BeginFenceEvent)
+DEFINE_EVENT_MATCHER(EndFenceEvent)
+DEFINE_EVENT_MATCHER(AllocateDataEvent)
+DEFINE_EVENT_MATCHER(DeallocateDataEvent)
+DEFINE_EVENT_MATCHER(CreateProfileSectionEvent)
+DEFINE_EVENT_MATCHER(DestroyProfileSectionEvent)
+DEFINE_EVENT_MATCHER(StartProfileSectionEvent)
+DEFINE_EVENT_MATCHER(StopProfileSectionEvent)
+DEFINE_EVENT_MATCHER(PushRegionEvent)
+DEFINE_EVENT_MATCHER(PopRegionEvent)
+DEFINE_EVENT_MATCHER(ProfileEvent)
+
+#define DEFINE_EVENT_WITH_NAME_MATCHER(__eventtype__)                                                             \
+template <typename Matcher>                                                                                       \
+auto A##__eventtype__##WithName(Matcher&& matcher)                                                                \
+{                                                                                                                 \
+    using EventType = Kokkos::utils::callbacks::__eventtype__;                                                    \
+    return ::testing::VariantWith<EventType>(::testing::Field(&EventType::name, std::forward<Matcher>(matcher))); \
+}
+
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelForEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelReduceEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelScanEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginFenceEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(CreateProfileSectionEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(PushRegionEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(ProfileEvent)
+
+namespace impl
+{
+
+template <typename T, typename... Matchers>
+class ContainsInOrderMatcher
+{
+public:
+    template <typename... Matchers_> requires std::conjunction_v<std::is_same<std::remove_cvref_t<Matchers_>, Matchers>...>
+    explicit ContainsInOrderMatcher(Matchers_&&... matchers_)
+      : matchers{std::forward<Matchers_>(matchers_)...} {}
+
+    template <typename IterableType>
+    bool MatchAndExplain([[maybe_unused]]const IterableType& arg, [[maybe_unused]]::testing::MatchResultListener* listener) const
+    {
+        if constexpr(sizeof...(Matchers) == 0) {
+            return true;
+        } else {
+            return MatchAndExplainImpl<0>(arg.cbegin(), arg.cend(), listener);
+        }
+    }
+
+    void DescribeTo(std::ostream* oss) const
+    {
+        *oss << "contains in order elements that match ";
+        DescribeInnerMatchersImpl(oss);
+    }
+
+    void DescribeNegationTo(std::ostream* oss) const
+    {
+        *oss << "does not contain in order elements that match ";
+        DescribeInnerMatchersImpl(oss);
+    }
+
+private:
+    template <size_t Idx, typename IteratorType>
+    bool MatchAndExplainImpl(IteratorType it_first, const IteratorType it_last, ::testing::MatchResultListener* listener) const
```

Comment:
Not sure but:

Reply:
Didn't work for the first `const` because the `operator<<` wants it to be non-const.